### ISSUE TITLE
Regexen on the LHS don't get syntax highlighted.

### ISSUE THREAD

**Issue by crazymykl** [atom/language-ruby#92]
I experienced this bug with the `=~` operator originally, but it's not limited to there.

**Comment by nn2242** [comment]
There's a few issues with regex expressions that I've seen: #82, #66. Also here are a couple of examples that may or may not be related.

Normally highlighted:
![screen shot 2015-05-10 at 10 53 40 am](https://cloud.githubusercontent.com/assets/2596124/7554771/4d5ae988-f703-11e4-9832-49dce8e8053d.png)

After adding a `then` keyword the regex loses the highlighting.
![screen shot 2015-05-10 at 10 53 21 am](https://cloud.githubusercontent.com/assets/2596124/7554770/48dbf640-f703-11e4-8640-104a66e19a76.png)

Another example of lack of regex highlighting combined with `then`:
![screen shot 2015-05-10 at 10 54 19 am](https://cloud.githubusercontent.com/assets/2596124/7554782/afc2e544-f703-11e4-81e3-b598f44621a7.png)

Completely breaks with `%` signs:
![screen shot 2015-05-10 at 10 54 06 am](https://cloud.githubusercontent.com/assets/2596124/7554785/d057b5be-f703-11e4-9752-be68206bcd25.png)


I'm not familiar with these highlighting rule systems for Atom but I'll try to dig in and see if there's a fix to be had somewhere.

**Comment by crazymykl** [comment]
Note that all your examples have the regex on the RHS, my issue is cases like:

```ruby
/\A\(?(?<area_code>\d{3})\)? ?(?<exchange_code>\d{3})[- ]?(?<subscriber_number>\d{4})\z/ =~ phone_number
```

Interestingly, that does get some syntax highlighting here.

**Comment by nn2242** [comment]
It looks like our issues are related after all. The problem is in the regex pattern defined here: https://github.com/atom/language-ruby/blob/master/grammars/ruby.cson#L499

The grammar currently prefers the division operator `/` over regex double slashes, so we're only matching a regex expression if the content following the double slashes is in a certain set or series of characters. `=~` and `then` are not currently in that list. You could add parentheses around your if condition or use the `%r` literal syntax to avoid the issue entirely.

Short term solution is to add the above mentioned operator and keyword to the whitelist, but even after adding those operators and keywords the pattern will still fail for things like `return /regex/ if true`.

**Comment by 50Wliu** [comment]
Looks like this was fixed by #96.

@nn2242 please correct me if I'm wrong.
